From bef8106946fa5436bf3429a99a0f2a2a90680a73 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Tue, 29 Oct 2019 07:58:50 +0200
Subject: [PATCH 25/32] cl-som-imx8: Fixed the PHYS_SDRAM_SIZE value for 1G
 configuration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixed the PHYS_SDRAM_SIZE value for 1G configuration.
It had an incorrect 2G value that was used as a parameter in the global
platform “imx8m_mem_map” structure in the DRAM descriptor.

Defined:
arch/arm/mach-imx/imx8m/soc.c:155

Used by:
arch/arm/cpu/armv8/cache_v8.c
setup_pgtables()

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/cl-som-imx8.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/cl-som-imx8.h b/include/configs/cl-som-imx8.h
index 96d7d99..c02e290 100644
--- a/include/configs/cl-som-imx8.h
+++ b/include/configs/cl-som-imx8.h
@@ -153,7 +153,13 @@
 
 #define CONFIG_SYS_SDRAM_BASE           0x40000000
 #define PHYS_SDRAM                      0x40000000
+
+#ifdef CONFIG_RAM_1G
+#define PHYS_SDRAM_SIZE			0x40000000 /* 1GB DDR */
+#else
 #define PHYS_SDRAM_SIZE			0x80000000 /* 2GB DDR */
+#endif
+
 #define CONFIG_NR_DRAM_BANKS		1
 
 #define CONFIG_SYS_MEMTEST_START    PHYS_SDRAM
-- 
1.9.1

